src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package taxi.service;
+
+import java.util.Optional;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByLogin(login);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Login or password are incorrect");
+        }
+        String driverPassword = driver.get().getPassword();
+        if (password.equals(driverPassword)) {
+            return driver.get();
+        }
+        throw new AuthenticationException("Login or password are incorrect");
```

Comment:
you can simplify this statement : if  !(driver.isEmpty()) && password.equals(driverPassword) return driver.get() else throw exception